recipe/build.sh

```diff
@@ -3,11 +3,8 @@
 CHOST=${macos_machine}
 
 FINAL_CPPFLAGS="-D_FORTIFY_SOURCE=2"
-FINAL_CFLAGS="-ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe"
-FINAL_CXXFLAGS="-ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0"
-if [[ "${version}" == "11.1.0" ]]; then
-  FINAL_CXXFLAGS="${FINAL_CXXFLAGS} -std=c++14"
-fi
+FINAL_CFLAGS="-ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe"
+FINAL_CXXFLAGS="-ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -stdlib=libc++ -isysroot \$CONDA_BUILD_SYSROOT -I\$PREFIX/include/c++/v1 -fvisibility-inlines-hidden -fmessage-length=0"
```

Comment:
```suggestion
FINAL_CXXFLAGS="-ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -stdlib=libc++ -isysroot \$CONDA_BUILD_SYSROOT -cxx-isystem\$PREFIX/include/c++/v1 -fvisibility-inlines-hidden -fmessage-length=0"
```

Could potentially use `--sysroot` here as well. It will update the search paths for everything to look in the given directory, not just for headers. Assuming there is no reason to mix things from the host system into our builds, this seems the easiest way to set paths across the board to our prefix.

`-cxx-isystem` over `-I` so we treat the contents of the `c++/v1` directory with higher priority than other non-system header paths.

https://clang.llvm.org/docs/ClangCommandLineReference.html

Reply:
I tried a few different iterations of `-cxx-isystem`, but unfortunately it resulted in the original problem, but `-I` works. 

Here's what I tried:

`-cxx-isystem $PREFIX/include/c++/v1`
`-cxx-isystem\$PREFIX/include/c++/v1`
`-cxx-isystem=$PREFIX/include/c++/v1`

Let me know if you spot what I might be doing wrong. I've left `-I` in there for now.

Reply:
Weird. Their docs specifically say that should work for C++ build environments. You weren't trying to build C code with that flag on as a test, were you? As far as I can tell, that's the only difference and that's what made me think `-cxx-system` was the right call for the C++-specific headers. As long as this does not impact building C code (main concern would be duplicate headers named the same thing and the C++ one gets used instead of the C one) then I don't see why `-I` shouldn't be fine as well.

Reply:
From what I read, SDKROOT should be enough to find the headers of the sysroot. It is set by our activation script. 
And -stdlib=libc++ should be enough to find the c++ headers.

Reply:
The doc also has -stdlib++-isystem https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang1-stdlib-isystem-directory

Reply:
What is the suggestion?

I posted a note here https://github.com/AnacondaRecipes/clang-compiler-activation-feedstock/pull/17#discussion_r2114202731

That is what happens when I remove what I added.

Reply:
Using `-stdlib++-isystem$PREFIX/include/c++/v1` instead of `-I$PREFIX/include/c++/v1`. If I understand the difference is that the libc++ headers will be treated as system headers. 

On the more broad topic I don't have one. Which downstream was showing the issue? Claude says this: https://claude.ai/share/a8350788-9a09-482f-86ab-07d9e55ee79e

Reply:
I can't see that link unfortunately. 

Both cctools-and-ld64 AND llvmdev showed the same/similar error messages.

Reply:
```
-stdlib++-isystem$PREFIX/include/c++/v1 
```

Fails when building cctools with:

```
/Users/lundby/anaconda/pr_feedback2/cctools-and-ld64_1748549346823/_build_env/include/c++/v1/cmath:321:5: error: <cmath> tried including <math.h> but didn't find libc++'s <math.h> header.           This usually means that your header search paths are not configured properly.           The header search paths should contain the C++ Standard Library headers before           any C Standard Library, and you are probably using compiler flags that make that           not be the case.
  321 | #   error <cmath> tried including <math.h> but didn't find libc++'s <math.h> header. \
      |     ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[4]: *** [Makefile:433: libParsers_la-archive_file.lo] Error 1
make[4]: Leaving directory '$SRC_DIR/cctools_build_final/ld64/src/ld/parsers'
make[3]: *** [Makefile:666: all-recursive] Error 1
make[3]: Leaving directory '$SRC_DIR/cctools_build_final/ld64/src/ld'
make[2]: *** [Makefile:383: all-recursive] Error 1
make[2]: Leaving directory '$SRC_DIR/cctools_build_final/ld64/src'
make[1]: *** [Makefile:384: all-recursive] Error 1
make[1]: Leaving directory '$SRC_DIR/cctools_build_final/ld64'
make: *** [Makefile:416: all-recursive] Error 1

BuildScriptException: Command '['/bin/bash', '-o', 'errexit', '/Users/lundby/anaconda/pr_feedback2/cctools-and-ld64_1748549346823/work/conda_build.sh']' returned non-zero exit status 2.

```

For my learnings, what is the concern over using `-I`?

Reply:
-isystem is best as the compiler does not emit warnings on what is flagged as system headers.